[flashlight-cuda] build failure

**Host Environment**
- OS: [ubuntu 20.1]
- Compiler: revision g++ 10.2.0

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install flashlight-cuda

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
```
   Working Directory: /home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/config-x64-linux-dbg-out.log
      /home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:328 (vcpkg_execute_required_process)
  ports/flashlight-cuda/portfile.cmake:30 (vcpkg_configure_cmake)
  scripts/ports.cmake:136 (include)
Error: Building package flashlight-cuda:x64-linux failed with: BUILD_FAILED
```
-Please attach any additional failure logs mentioned in the console output.
```
In file included from /home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/nn/modules/Conv2D.h:10,
                 from /home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/nn/modules/Conv2D.cpp:8:
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:137:45: error: ‘string’ in namespace ‘std’ does not name a type
  137 |   static OptimLevel toOptimLevel(const std::string& in);
      |                                             ^~~~~~
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:12:1: note: ‘std::string’ is defined in header ‘<string>’; did you forget t
o ‘#include <string>’?
   11 | #include <unordered_map>
  +++ |+#include <string>
   12 | 
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:40: error: ‘string’ is not a member of ‘std’
  139 |   static const std::unordered_map<std::string, OptimLevel> kStringToOptimLevel;
      |                                        ^~~~~~
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:40: note: ‘std::string’ is defined in header ‘<string>’; did you forget
 to ‘#include <string>’?
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:40: error: ‘string’ is not a member of ‘std’
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:40: note: ‘std::string’ is defined in header ‘<string>’; did you forget
 to ‘#include <string>’?
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:58: error: template argument 1 is invalid
  139 |   static const std::unordered_map<std::string, OptimLevel> kStringToOptimLevel;
      |                                                          ^
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:58: error: template argument 3 is invalid
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:58: error: template argument 4 is invalid
/home/ngoel-00171/vcpkg/buildtrees/flashlight-cuda/src/4bc14022f7-7b8557c63e.clean/flashlight/fl/common/Defines.h:139:58: error: template argument 5 is invalid
```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
export CUDACXX=/usr/local/cuda/bin/nvcc

--- a/ports/flashlight-cuda/portfile.cmake
+++ b/ports/flashlight-cuda/portfile.cmake
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO facebookresearch/flashlight
-    REF 94486a1108fd0511a05523bfa1cf734bc14022f7
-    SHA512 73a7547a26020be21d26393c2758cc235b452a0fb5d8537777bdbcc556f71846e07eca649167902dd7d0743cb7d10738b4e38cd7fe69b51b6122f3241e2edd84
+    REF 0948e97dbff23d474500a5f55bcf59d3b3589cf4
+    SHA512 89c2efdebc6688a38f9fc3ebb8442c8fe2ad3d51c0e60c5dd8e7ca283715e07f48083543f09eed0db9929d86b765a898d9b3960a9b6f9844b20260f94cf71fdf
     HEAD_REF master
 )
 
@@ -14,23 +14,24 @@ set(FL_DEFAULT_VCPKG_CMAKE_FLAGS
   -DFL_BACKEND=CUDA # this port is CUDA-backend only
   -DFL_BUILD_STANDALONE=OFF
   -DFL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/share/${PORT} # for CMake configs/targets
-  )
+)
 
-# Determine which backend to build via specified feature
+# Determine which components to build via specified feature
 vcpkg_check_features(
-  OUT_FEATURE_OPTIONS FL_BACKEND_FEATURE_OPTIONS
+  OUT_FEATURE_OPTIONS FL_FEATURE_OPTIONS
   FEATURES
     lib FL_BUILD_LIBRARIES
     fl FL_BUILD_CORE
     asr FL_BUILD_APP_ASR
-    imgclass FL_BUILD_APP_IMG_CLASS
+    imgclass FL_BUILD_APP_IMGCLASS
+    lm FL_BUILD_APP_LM
 )
 
 # Build and install
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
-    OPTIONS ${FL_DEFAULT_VCPKG_CMAKE_FLAGS} ${FL_BACKEND_FEATURE_OPTIONS}
+    OPTIONS ${FL_DEFAULT_VCPKG_CMAKE_FLAGS} ${FL_FEATURE_OPTIONS}
 )
 vcpkg_install_cmake()
 
@@ -40,16 +41,24 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 # Binaries/tools
 set(FLASHLIGHT_TOOLS "")
 if ("imgclass" IN_LIST FEATURES)
-  list(APPEND FLASHLIGHT_TOOLS fl_imageNetResnet34)
+  list(APPEND FLASHLIGHT_TOOLS fl_img_imagenet_resnet34)
 endif()
 if ("asr" IN_LIST FEATURES)
-  list(APPEND FLASHLIGHT_TOOLS fl_asr_train fl_asr_test fl_asr_decode)
+  list(APPEND FLASHLIGHT_TOOLS
+    fl_asr_train
+    fl_asr_test
+    fl_asr_decode
+    fl_asr_align
+    fl_asr_voice_activity_detection_ctc
+  )
+endif()
+if ("lm" IN_LIST FEATURES)
+  list(APPEND FLASHLIGHT_TOOLS fl_lm_train fl_lm_dictionary_builder)
 endif()
 list(LENGTH FLASHLIGHT_TOOLS NUM_TOOLS)
 if (NUM_TOOLS GREATER 0)
   vcpkg_copy_tools(TOOL_NAMES ${FLASHLIGHT_TOOLS} AUTO_CLEAN)
 endif()
 
-# Copyright and license
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/flashlight-cuda RENAME copyright)
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/flashlight-cuda RENAME license)
+# Copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

--- a/ports/flashlight-cuda/vcpkg.json
+++ b/ports/flashlight-cuda/vcpkg.json
@@ -2,6 +2,7 @@
   "name": "flashlight-cuda",
   "version-string": "20201201",
   "description": "A C++ standalone library for machine learning. CUDA backend.",
+  "homepage": "https://github.com/facebookresearch/flashlight",
   "supports": "!(windows | osx)",
   "default-features": [
     "fl"
@@ -12,9 +13,9 @@
       "dependencies": [
         {
           "name": "flashlight-cuda",
-          "default-features": false,
           "features": [
-            "fl"
+            "fl",
+            "lib"
           ]
         },
         "gflags",
@@ -34,13 +35,6 @@
         "cereal",
         "cuda",
         "cudnn",
-        {
-          "name": "flashlight-cuda",
-          "default-features": false,
-          "features": [
-            "lib"
-          ]
-        },
         "nccl",
         "openmpi",
         "stb"
@@ -51,9 +45,9 @@
       "dependencies": [
         {
           "name": "flashlight-cuda",
-          "default-features": false,
           "features": [
-            "fl"
+            "fl",
+            "lib"
           ]
         },
         "gflags"
@@ -66,6 +60,19 @@
         "fftw3",
         "intel-mkl",
         "kenlm"
+      ]
+    },
+    "lm": {
+      "description": "flashlight lm app",
+      "dependencies": [
+        {
+          "name": "flashlight-cuda",
+          "features": [
+            "fl",
+            "lib"
+          ]
+        },
+        "gflags"
       ]
     }
   }

